bin/webpack-configs.js

```diff
@@ -260,32 +260,16 @@ const getMainConfig = ( options = {} ) => {
 			new CopyWebpackPlugin( {
 				patterns: [
 					{
-						from: './assets/js/blocks/checkout/block.json',
-						to: './checkout/block.json',
-					},
-					{
-						from:
-							'./assets/js/blocks/featured-items/featured-category/block.json',
-						to: './featured-category/block.json',
-					},
-					{
-						from:
-							'./assets/js/blocks/featured-items/featured-product/block.json',
-						to: './featured-product/block.json',
-					},
-					{
-						from:
-							'./assets/js/blocks/handpicked-products/block.json',
-						to: './handpicked-products/block.json',
-					},
-					{
-						from: './assets/js/blocks/product-tag/block.json',
-						to: './product-tag/block.json',
-					},
-					{
-						from:
-							'./assets/js/blocks/products-by-attribute/block.json',
-						to: './products-by-attribute/block.json',
+						from: './assets/js/blocks/**/block.json',
+						to( { absoluteFilename } ) {
+							const blockName = absoluteFilename
+								.split( '/' )
+								.at( -2 );
+							return `./${ blockName }/block.json`;
+						},
+						globOptions: {
+							ignore: [ '**/inner-blocks/**' ],
+						},
```

Comment:
To have a cleaner (and less conflict) folder structure, we might want to pivot away from `${ blockName }/block.json` toward `${ blockName }.block.json` instead, but I'm not sure how would that play into translation.

> `ignore: [ '**/inner-blocks/**' ],`

I believe lack of having `assets` in our releases (not confirmed?) and not having them in `build` is the reason why block metadata translation isn't loading in the editor :/

Reply:
> I believe lack of having assets in our releases (not confirmed?) and not having them in build is the reason why block metadata translation isn't loading in the editor :/

We only need `block.json` files in the `build` to translate the block metadata. `asset` is unnecessary to get translation working in my test.

> we might want to pivot away from ${ blockName }/block.json toward ${ blockName }.block.json instead, but I'm not sure how would that play into translation.

I will test this. But it's possible that `make-pot` scans only `block.json` files.

Reply:
> I will test this. But it's possible that `make-pot` scans only `block.json` files.

It is. See https://github.com/wp-cli/i18n-command/blob/main/src/MakePotCommand.php#L666-L679